rhc-ose-ansible/roles/post_install/tasks/main.yaml

```diff
@@ -0,0 +1,45 @@
+---
+
+  - name: "backup image streams"
+    command: cp /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.1/image-streams/image-streams-rhel7.json /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.1/image-streams/image-streams-rhel7.json.orig
+    when: target_registry is defined
+
+  - name: "Configure ImageStreams"
+    command: sed -i -e 's/registry.access.redhat.com/{{ target_registry }}/g' /usr/share/ansible/openshift-ansible/roles/openshift_examples/files/examples/v1.1/image-streams/image-streams-rhel7.json
+    when: target_registry is defined
+
+  - name: "Check for existing registry"
+    command: oc get dc docker-registry -n default
+    register: registry_output
+    ignore_errors: yes
+    failed_when: "'FAILED' in registry_output.stderr"
+
+  - name: "Checking for existing router"
+    command: oc get dc router -n default
+    register: router_output
+    failed_when: "'FAILED' in router_output.stderr"
+
+  - name: "Checking for existing ImageStreams"
+    command: oc get is php -n openshift
+    register: is_output
+    failed_when: "'FAILED' in is_output.stderr"
+
+  - name: "setup ose docker registry with images from registry other than redhat"
+    shell: oadm registry --config=/etc/origin/master/admin.kubeconfig --credentials=/etc/origin/master/openshift-registry.kubeconfig --images='{{ target_registry }}/openshift3/ose-${component}:${version}' -n default
+    when: target_registry is defined and registry_output.rc != 0
```

Comment:
If you setup openshift_docker_additional_registries (and openshift_docker_insecure_registries if the registry is insecure) you can skip the --images override here, as it will attempt to pull from the additional registry before escalating to registry.access.redhat.com. (The variables openshift_docker_* are only available in master and pre-release 3.2 builds currently, they are named cli_docker_* in earlier releases of atomic-openshift-installer).